node/node_newblock.go

```diff
@@ -71,10 +72,15 @@ func (node *Node) WaitForConsensusReadyv2(readySignal chan struct{}, stopChan ch
 					if len(node.pendingTransactions) < threshold && time.Now().Before(deadline) {
 						continue
 					}
+
+					coinbase := node.Consensus.SelfAddress
 					// Normal tx block consensus
-					selectedTxs := node.getTransactionsForNewBlock(MaxNumberOfTransactionsPerBlock)
+					selectedTxs := types.Transactions{} // Empty transaction list
+					if node.NodeConfig.GetNetworkType() != nodeconfig.Mainnet {
+						selectedTxs = node.getTransactionsForNewBlock(MaxNumberOfTransactionsPerBlock, coinbase)
+					}
 					utils.GetLogInstance().Info("PROPOSING NEW BLOCK ------------------------------------------------", "blockNum", node.Blockchain().CurrentBlock().NumberU64()+1, "threshold", threshold, "selectedTxs", len(selectedTxs))
-					if err := node.Worker.CommitTransactions(selectedTxs); err != nil {
+					if err := node.Worker.CommitTransactions(selectedTxs, coinbase); err != nil {
```

Comment:
If set coinbase at node.getTransactionsForNewBlock, here we don't have to set coinbase again? Or we can set everything here instead of at getTransactionsForNewBlock

Reply:
One is for update headers, another is for processing transactions. Both needed.